internal/crypto/srp.go

```diff
@@ -0,0 +1,248 @@
+package crypto
+
+import (
+	"crypto/sha512"
+	"io"
+	"math/big"
+
+	"golang.org/x/crypto/pbkdf2"
+	"golang.org/x/xerrors"
+
+	"github.com/gotd/xor"
+)
+
+type SRP struct {
```

Comment:
There should be definition for "SRP" somewhere.

Probably we can even rename it to something longer, but more common